Architecture.md

```diff
@@ -4,79 +4,150 @@
 
 ### Description and intent
 
-***TODO: [UNIFIED WORK] write description of the software and its intent (what market needs it answers to)***
+**_TODO: [UNIFIED WORK] write description of the software and its intent (what market needs it answers to)_**
 
 ### Technologies
 
-***TODO: [UNIFIED WORK] define the different technologies below, why they were chosen, and use arguments AND PROOFS to show research***
+Below are the technologies used for the different components of the application. We will discuss the choices that we made, and why the chosen technology fits in the ecosystem.
+
+A list of TPMs (Third-Party Modules) might be written to give more insight about the component, but please keep in mind those lists are non-exhaustive and only the main TPMs are described.
+
+**_TODO: [UNIFIED WORK] define the different technologies below, why they were chosen, and use arguments AND PROOFS to show research_**
 
 #### Web application
 
-***TODO: [CLIENT] fill for web application frontend***
+**_TODO: [CLIENT] fill for web application frontend_**
 
 #### CLI (Command-Line Interface)
 
-***TODO: [CLIENT] fill for the cli***
+**_TODO: [CLIENT] fill for the cli_**
 
 #### Client API
 
-***TODO: [CLIENT] fill for the external API***
+
+The Client API is one of the fundamental parts of our project. It not only handles all Client interactions, but is also the only application that is writing any data in the shared database. It notifies Pomegranate each time a project needs to be deployed and lets a user add his SSH keys to be able to connect to the local Git server. Every user request is send and verified by the API before being carried out.
```

Comment:
We can use "their" instead of "his" to make it more gender neutral ;) Also, does the user already know what is Pomegranate in this section? Maybe we should add some cross reference or footnote? Like "(cf. chapter 3.2).
```suggestion
The Client API is one of the fundamental parts of our project. It not only handles all Client interactions, but is also the only application that can write data in the shared database. It notifies Pomegranate each time a project needs to be deployed and lets a user add their SSH keys to be able to connect to the local Git server. Every user request is send and verified by the API before being carried out.
```